test/new-e2e/tests/cspm/cspm_test.go

```diff
@@ -0,0 +1,298 @@
+	"cis-kubernetes-1.5.1-4.2.1": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.3": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.4": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.5": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.6": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.10": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.12": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+}
+
+func TestCSPM(t *testing.T) {
+	values := `
+datadog:
+  securityAgent:
+    compliance:
+      enabled: true`
+
+	e2e.Run(t, &cspmTestSuite{}, e2e.WithProvisioner(localkubernetes.Provisioner(localkubernetes.WithAgentOptions(kubernetesagentparams.WithHelmValues(values), kubernetesagentparams.WithoutDualShipping()))))
+}
+
+func (s *cspmTestSuite) TestFindings() {
+	res, err := s.Env().KubernetesCluster.Client().CoreV1().Pods("datadog").List(context.Background(), metav1.ListOptions{
+		LabelSelector: fields.OneTermEqualSelector("app", s.Env().Agent.LinuxNodeAgent.LabelSelectors["app"]).String(),
+	})
+	assert.NoError(s.T(), err)
+	assert.Len(s.T(), res.Items, 1)
+	agentPod := res.Items[0]
+	_, _, err = s.Env().KubernetesCluster.KubernetesClient.PodExec("datadog", agentPod.Name, "security-agent", []string{"security-agent", "compliance", "check", "--dump-reports", "/tmp/reports", "--report"})
+	assert.NoError(s.T(), err)
+	dumpContent, _, err := s.Env().KubernetesCluster.KubernetesClient.PodExec("datadog", agentPod.Name, "security-agent", []string{"cat", "/tmp/reports"})
+	assert.NoError(s.T(), err)
+	findings, err := parseFindingOutput(dumpContent)
+	assert.NoError(s.T(), err)
+	s.checkFindings(findings, mergeFindings(expectedFindingsMasterEtcdNode, expectedFindingsWorkerNode))
+}
+
+func (s *cspmTestSuite) TestMetrics() {
+	s.T().Log("Waiting for datadog.security_agent.compliance.running metrics")
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+
+		metrics, err := s.Env().FakeIntake.Client().FilterMetrics("datadog.security_agent.compliance.running")
+		assert.NoError(c, err)
+		assert.NotEmpty(c, metrics)
+		s.T().Log("Metrics found: datadog.security_agent.compliance.running")
+	}, 2*time.Minute, 10*time.Second)
+
+	s.T().Log("Waiting for datadog.security_agent.compliance.containers_running metrics")
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		metrics, err := s.Env().FakeIntake.Client().FilterMetrics("datadog.security_agent.compliance.containers_running")
+		assert.NoError(c, err)
+		assert.NotEmpty(c, metrics)
+		s.T().Log("Metrics found: datadog.security_agent.compliance.containers_running")
+	}, 2*time.Minute, 10*time.Second)
+
+}
+func (s *cspmTestSuite) checkFindings(findings, expectedFindings findings) {
+	checkedRule := []string{}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Would add `t.Helper()` here so that this logs with the caller rather than with this line